src/Agent.php

```diff
@@ -39,16 +39,18 @@ public function registerCollectors(Application $app): void
     {
         if (config('elastic-apm-laravel.spans.querylog.enabled') !== false) {
             // DB Queries collector
-            $this->collectors->put(
-                DBQueryCollector::getName(),
-                new DBQueryCollector($app, $this->request_start_time)
-            );
+            $this->addCollector(new DBQueryCollector($app, $this->request_start_time));
         }
 
         // Http request collector
+        $this->addCollector(new HttpRequestCollector($app, $this->request_start_time));
+    }
+
+    public function addCollector(DataCollectorInterface $collector)
+    {
         $this->collectors->put(
-            HttpRequestCollector::getName(),
-            new HttpRequestCollector($app, $this->request_start_time)
+            $collector->getName(),
```

Comment:
Does this work, taking into account that the method is defined as static?

Reply:
I had the same question! So I double-checked in tinker before I pushed it up. Works both ways.
```
>>> $x = new AG\ElasticApmLaravel\Collectors\DBQueryCollector(app(), 1.0);
=> AG\ElasticApmLaravel\Collectors\DBQueryCollector {#3879}
>>> echo $x->getName();
query-collector⏎
>>> echo $x::getName();
query-collector⏎
>>> exit
```

Reply:
In that case, maybe we should change the interface, it doesn't make sense anymore to have a `static` method definition. But since it is a breaking change, let's create a new interface following Laravel's folder and naming structure: `AG\ElasticApmLaravel\Contracts\DataCollector`